src/ConstructiveSolidGeometry/VolumePrimitives/VolumePrimitives.jl

```diff
@@ -1,5 +1,19 @@
-include("Cone.jl")
+ClosedPrimitive(p::VP) where {VP <:AbstractVolumePrimitive} = VP(p, COT = ClosedPrimitive)
+OpenPrimitive(p::VP) where {VP <: AbstractVolumePrimitive} = VP(p, COT = OpenPrimitive)
+switchClosedOpen(p::AbstractVolumePrimitive{<:Any,ClosedPrimitive}) = OpenPrimitive(p)
+switchClosedOpen(p::AbstractVolumePrimitive{<:Any,OpenPrimitive}) = ClosedPrimitive(p)
+
+distance(pt::CartesianPoint, vp::AbstractVolumePrimitive) = 
+minimum(map(p -> distance(pt, p), surfaces(vp)))
```

Comment:
`map` or `broadcast`?

Reply:
Did not thought about it or tested it yet. 
We will probably handle this anyhow a bit different in the charge drift.